src/test/app/AccountTxPaging_test.cpp

```diff
@@ -16,243 +16,259 @@
+
+        return env.rpc("json", "account_tx", to_string(jvc))[jss::result];
     }
 
     void
     testAccountTxPaging ()
     {
-        using namespace std::placeholders;
-
-        bool const forward = true;
-
-        std::int32_t min_ledger;
-        std::int32_t max_ledger;
-        Json::Value token;
-        int limit;
-
-        // the supplied account-tx-transactions.db contains contains
-        // transactions with the following ledger/sequence pairs.
-        //  3|5
-        //  4|4
-        //  4|10
-        //  5|4
-        //  5|7
-        //  6|1
-        //  6|5
-        //  6|6
-        //  6|7
-        //  6|8
-        //  6|9
-        //  6|10
-        //  6|11
-
-        min_ledger = 2;
-        max_ledger = 5;
-
-        {
-            limit = 2;
+        testcase("Paging for Single Account");
+        using namespace test::jtx;
 
-            BEAST_EXPECT(next(limit, forward, token, min_ledger, max_ledger) == 2);
-            checkTransaction (txs_[0], 3, 5);
-            checkTransaction (txs_[1], 4, 4);
-            checkToken (token, 4, 10);
+        Env env(*this);
+        Account A1 {"A1"};
+        Account A2 {"A2"};
+        Account A3 {"A3"};
 
-            BEAST_EXPECT(next(limit, forward, token, min_ledger, max_ledger) == 2);
-            checkTransaction (txs_[0], 4, 10);
-            checkTransaction (txs_[1], 5, 4);
-            checkToken (token, 5, 7);
+        env.fund(XRP(10000), A1, A2, A3);
+        env.close();
 
-            BEAST_EXPECT(next(limit, forward, token, min_ledger, max_ledger) == 1);
-            checkTransaction (txs_[0], 5, 7);
-
-            BEAST_EXPECT(! token["ledger"]);
-            BEAST_EXPECT(! token["seq"]);
-        }
-
-        token = Json::nullValue;
-
-        min_ledger = 3;
-        max_ledger = 9;
+        env.trust(A3["USD"](1000), A1);
+        env.trust(A2["USD"](1000), A1);
+        env.trust(A3["USD"](1000), A2);
+        env.close();
 
+        int count = 0;
+        while (count < 5)
         {
-            limit = 1;
-
-            BEAST_EXPECT(next(limit, forward, token, min_ledger, max_ledger) == 1);
-            checkTransaction (txs_[0], 3, 5);
-            checkToken (token, 4, 4);
-
-            BEAST_EXPECT(next(limit, forward, token, min_ledger, max_ledger) == 1);
-            checkTransaction (txs_[0], 4, 4);
-            checkToken (token, 4, 10);
-
-            BEAST_EXPECT(next(limit, forward, token, min_ledger, max_ledger) == 1);
-            checkTransaction (txs_[0], 4, 10);
-            checkToken (token, 5, 4);
+            env(pay(A2, A1, A2["USD"](2)));
+            env(pay(A3, A1, A3["USD"](2)));
+            env(offer(A1, XRP(11), A1["USD"](1)));
+            env(offer(A2, XRP(10), A2["USD"](1)));
+            env(offer(A3, XRP(9),  A3["USD"](1)));
+            env.close();
+            count++;
         }
 
+        /* The sequence/ledger for A3 are as follows:
+         * seq     ledger_index
+         * 3  ----> 3
```

Comment:
so, I did in-fact get the complete list of txs for this account by making an unlimited call to the same API and then I made the various paging scenarios match that master list. As you say, this is pretty much how the original test was, although it was based on a rather opaque transaction DB, so I like this slightly better - at least you can try to correlate each `env.close()` with an increment in the ledger number.  I'll leave this as is for now and can revisit in the next sweep of test tasks wherein we try to add more coverage.